### ISSUE TITLE
`import-error` based on `ignored-modules` is inconsistent for wildcards

### ISSUE THREAD

**Issue by aaron-skydio** [PyCQA/pylint#8232]
### Bug description

# pylint: disable=missing-module-docstring,unused-import
import foo

### Configuration

```ini
ignored-modules=fo*
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:2:0: E0401: Unable to import 'foo' (import-error)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

Expected behavior is no errors here.

The error goes away if using `ignored-modules=foo` instead of `ignored-modules=fo*`.  This seems to be because the logic [here](https://github.com/PyCQA/pylint/blob/9490a38b62d5c8750d74a186b644e060de8313d2/pylint/checkers/imports.py#L148) checks `ignored-modules`, but does not invoke `fnmatch`, and should probably be changed to invoke `fnmatch`?

### Pylint version

```shell
pylint 2.16.1
astroid 2.14.1
Python 3.8.14 (default, Sep  8 2022, 00:02:10)
```
